xplique/plots/craft_visualizations.py

```diff
@@ -0,0 +1,297 @@
+"""
+Pretty plots for Craft, framework agnostic (Tensorflow / Pytorch)
+"""
+
+import numpy as np
+import cv2
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap, LinearSegmentedColormap
+import matplotlib.gridspec as gridspec
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+import colorsys
+from math import ceil
+
+class CraftPlot():
+    """
+    Class to display various Craft plots & attribution maps
+    
+    Parameters
+    ----------
+        crops
+            The crops (X in the paper)
+        crops_u
+            The concepts values (U)
+        importances
+            The Sobol total index (importance score) for each concept.
+        nb_most_important_concepts
+            The number of concepts to focus on. Default is 5.
+    """
+    def __init__(self,
+                 crops: np.ndarray,
+                 crops_u: np.ndarray,
+                 importances: list,
+                 nb_most_important_concepts: int = 6):
+        self.crops = crops
+        self.crops_u = crops_u
+        self.importances = importances
+
+        self.cmaps = None
+
+        self.most_important_concepts = np.argsort(self.importances)[::-1][:nb_most_important_concepts]
+        
+        if nb_most_important_concepts > len(self.importances):
+            raise RuntimeError('nb_most_important_concepts should be <= len of importances')
+
+    def plot_concepts_importances(self, only_most_important: bool=False, quiet: bool=False):
+        """
+        Plot a bar chart displaying the importance value of each concept.
+        """
+        if not only_most_important:
+            plt.bar(range(len(self.importances)), self.importances)
+            plt.xticks(range(len(self.importances)))
+            plt.title("Concept Importance")
+        else:
+            if self.cmaps == None:
+                self.set_concept_attribution_cmap()
+            importances = self.importances[self.most_important_concepts]
+            colors = [c(1.0) for c in self.cmaps]
+            plt.bar(range(len(importances)), importances, color=colors)
+            plt.xticks(range(len(importances)))
+            plt.title("Concept Importance")
+
+        if not quiet:
+            for c_id in self.most_important_concepts:
+                print("Concept", c_id, " has an importance value of ", self.importances[c_id])
+
+    @staticmethod
+    def _show(img, **kwargs):
+        img = np.array(img)
+        if img.shape[0] == 3:
+            img = img.transpose(1, 2, 0)
+
+        img -= img.min()
+        if img.max() > 0:
+            img /= img.max()
+        plt.imshow(img, **kwargs); plt.axis('off')
+
+    def plot_concepts_crops(self, nb_crops: int = 10) -> None:
+        """
+        Display the crops for each concept.
+
+        Parameters
+        ----------
+            nb_crops
```

Comment:
Typo

Reply:
Sorry but I am not sure what is the typo here ? (maybe it's somewhere else ?)

Reply:
Sorry but I am not sure what is the typo here ? (maybe it's somewhere else ?)